.github/generate-quest-metadata.main.kts

```diff
@@ -0,0 +1,505 @@
+        // Add both variants
+        allChanges.add(TaginfoChange(key, "yes", change))
+        allChanges.add(TaginfoChange(key, "no", change))
+      } else if (Regex("\\.toString$").containsMatchIn(value) || Regex("\\.joinToString\\(").containsMatchIn(value)) {
+        // No value to add
+        // TODO: Check what value we should be setting
+        allChanges.add(TaginfoChange(key, "", change))
+      } else if (Regex("^when\\(answer").containsMatchIn(value)) {
+        // TODO: Unpick this
+        println(value + " is unknown when")
+        //allChanges.add(TaginfoChange(key, "", change))
+      } else {
+        if (!stringCheck.matches(value)) {
+          println("Value '" + value + "' is not a string")
+          if (value.startsWith("if (")) {
+            // TODO: Fix me!
+            println("Skipping if statement")
+            allChanges.add(TaginfoChange(key, "", change))
+            continue
+          }
+          var questAnswerType = getQuestAnswerType(questFileContent)
+          println("Got quest answer types: " + questAnswerType)
+          for (it in questAnswerType) {
+            println("Got quest answer type: " + it)
+            if (it == "Boolean" || it == "String" || it == "Unit") {
+              println("Skipping generic " + it + "...")
+              allChanges.add(TaginfoChange(key, "", change))
+              continue
+            } else if (Regex("^List<").containsMatchIn(it)) {
+              println("Skipping generic List...")
+              allChanges.add(TaginfoChange(key, "", change))
+              continue
+            } else if (it == "SuspectedOnewayAnswer" || it == "CyclewayAnswer" || it == "SidewalkAnswer" || it == "ShopTypeAnswer" || it == "CompletedConstructionAnswer") {
+              println("Skipping, it's complicated " + it + "...")
+              allChanges.add(TaginfoChange(key, "", change))
+              continue
+            } else if (it == "RecyclingContainerMaterialsAnswer" || it == "SurfaceAnswer" || it == "MaxSpeedAnswer" || it == "MaxHeightAnswer" || it == "LanesAnswer") {
+              // TODO: Fix me!
+              println("Skipping, needs investigation " + it + "...")
+              allChanges.add(TaginfoChange(key, "", change))
+              continue
+//            } else if (it == "BuildingType" || it == "DrinkingWater" || it == "PoliceType") {
+//              // TODO: Fix me!
+//              println("Skipping, complex enum " + it + "...")
+//              allChanges.add(TaginfoChange(key, "", change))
+//              continue
+            } else if (it == "BusStopShelterAnswer" || it == "StepsRampAnswer") {
+              // TODO: Fix me!
+              println("Skipping, simple enum " + it + "...")
+              allChanges.add(TaginfoChange(key, "", change))
+              continue
+            } else if (it == "RoadNameAnswer" || it == "PlaceNameAnswer" || it == "BusStopRefAnswer" || it == "HousenumberAnswer" || it == "PostboxRefAnswer") {
```

Comment:
Just randomly looking through this PR...

```suggestion
            } else if (it in setOf("RoadNameAnswer","PlaceNameAnswer","BusStopRefAnswer","HousenumberAnswer","PostboxRefAnswer") {
```

Though that the script relies on hardcoded quest names is maybe a red flag anyway

Reply:
Thanks for the hint, I've applied it there and in a few other places.

> Though that the script relies on hardcoded quest names is maybe a red flag anyway

See discussion here as to why they're (currently) hardcoded:
https://github.com/streetcomplete/StreetComplete/pull/2754#issuecomment-901544052